In [10]:
import json
import numpy as np
import pandas as pd

In [214]:
def array(a):
    return np.array(a)

with open('res_TokenNumeral_test.v6.json','r') as f:
    s = f.read()
    s = eval(s)
    print(s)

{0: {'x': [3, 10, 15, 25, 40, 100, 150, 200, 250, 320], 'topk': [[0.0075, 0.025, 0.0375, 0.0625, 0.1, 0.25, 0.375, 0.5, 0.625, 0.8], [0.0825, 0.16, 0.1975, 0.265, 0.3575, 0.58, 0.7375, 0.8275, 0.915, 0.985], [0.0575, 0.1125, 0.16, 0.2375, 0.3175, 0.57, 0.7125, 0.835, 0.9225, 0.9825], [0.01, 0.0275, 0.0375, 0.0575, 0.1025, 0.245, 0.3775, 0.7325, 0.8825, 0.9875], [0.0075, 0.0275, 0.035, 0.045, 0.0975, 0.2525, 0.3775, 0.66, 0.8475, 0.9775]], 'mape': [11176.740968133583, 3252.972592833517, 14866329.05507188, 16236152.632706832], 'mae': [520221961.2086511, 520074703.5397402, 533997887.8772472, 534942880.25352484], 'mdae': [767.5, 1138.5, 20138181.0, 20138422.0], 'mdape': [0.9821992691535801, 0.9838698421201435, 10279.818401221572, 11605.111661239906], 'RMSE': [10308170422.02438, 10308170445.96279, 10307169210.032585, 10307170325.909687], 'avg_rank': [200.5, 100.5425, 103.6075, 155.9925, 163.2225]}, 1: {'x': [3, 10, 15, 25, 40, 100, 150, 200, 250, 320], 'topk': [[0.0075, 0.025, 0.0375, 0.062

In [216]:
res = {}
res_list = []
score = 5 # 1,2,3,4,5=r
# v = s
for score in [2]:
    
    for k, v in s.items():
        print(k)
#         if k != '500':
#             continue

        avg_rank = np.array(v['avg_rank'])
        if score == 5:
            avg_avg_rank = (np.mean(avg_rank[1:]))
        else:
            avg_avg_rank = (avg_rank[1:][score - 1])

        print(avg_avg_rank)

        mae = np.array(v['mae'])
        if score == 5:
            avg_mae = (np.mean(mae))
        else:
            avg_mae = (mae[score - 1])

        print(avg_mae)

        mdae = np.array(v['mdae'])
        if score == 5:
            avg_mdae = (np.mean(mdae))
        else:
            avg_mdae = mdae[score -1]
        print(avg_mdae)

        mape = np.array(v['mape'])
        if score == 5:
            avg_mape = (np.mean(mape))
        else:
            avg_mape = mape[score -1]

        print(avg_mape)

        mdape = np.array(v['mdape'])
        if score == 5:
            avg_mdape = (np.mean(mdape))
        else:
            avg_mdape = mdape[score -1]
        print(avg_mdape)

        RMSE = np.array(v['RMSE'])
        if score == 5:
            avg_RMSE = (np.mean(RMSE))
        else:
            avg_RMSE = RMSE[score -1]
        print(avg_RMSE)

        lll = [avg_avg_rank, avg_mae, avg_mdae, avg_mape, avg_mdape, avg_RMSE]
        print(lll)
        lll = [i[0] if type(i) == type(np.array([1])) else i for i in lll]
        print(lll)
        lll = [round(i, 2) for i in lll]
        res[k] = lll
        res_list.append(lll)
        
print(np.mean(np.array(res_list), axis=0))
        

0
103.6075
520074703.5397402
1138.5
3252.972592833517
0.9838698421201435
10308170445.96279
[103.6075, 520074703.5397402, 1138.5, 3252.972592833517, 0.9838698421201435, 10308170445.96279]
[103.6075, 520074703.5397402, 1138.5, 3252.972592833517, 0.9838698421201435, 10308170445.96279]
1
103.6075
520074703.5397402
1138.5
3252.972592833517
0.9838698421201435
10308170445.96279
[103.6075, 520074703.5397402, 1138.5, 3252.972592833517, 0.9838698421201435, 10308170445.96279]
[103.6075, 520074703.5397402, 1138.5, 3252.972592833517, 0.9838698421201435, 10308170445.96279]
2
103.6075
520074703.5397402
1138.5
3252.972592833517
0.9838698421201435
10308170445.96279
[103.6075, 520074703.5397402, 1138.5, 3252.972592833517, 0.9838698421201435, 10308170445.96279]
[103.6075, 520074703.5397402, 1138.5, 3252.972592833517, 0.9838698421201435, 10308170445.96279]
3
103.6075
520074703.5397402
1138.5
3252.972592833517
0.9838698421201435
10308170445.96279
[103.6075, 520074703.5397402, 1138.5, 3252.972592833517, 0.9

# size val 40000000

In [115]:
pd.DataFrame.from_dict(res, orient='index', columns=['avg_rank', 'mae', 'mdae', 'mape', 'mdape', 'RMSE'])

avg_rank       mae     mdae       mape  mdape       RMSE
50      24.35  40490.85  1753.86  121752.71   1.00  113381.07
100     23.87  13900.20  1330.50    1688.76   0.86   53755.74
200     22.62  12873.01   196.00     510.56   0.49   49865.61
300     20.81   9292.91  1493.77    1226.75   0.87   47651.26
500     22.84  13445.14   274.00    1416.25   0.60   65193.08

# alpha val 40000000

In [112]:
pd.DataFrame.from_dict(res, orient='index', columns=['avg_rank', 'mae', 'mdae', 'mape', 'mdape', 'RMSE'])

avg_rank       mae     mdae     mape  mdape      RMSE
0.5      21.56   9649.68  2147.27  2099.02   0.97  47956.03
0.75     21.09   9507.21  1807.27  2089.75   0.95  48040.27
1        20.81   9292.91  1493.77  1226.75   0.87  47651.26
1.25     21.02  13686.14   345.00  1101.63   0.81  51712.54
1.5      20.74   8287.48   180.50   163.87   0.42  46247.25
2.5      21.39  14604.43   253.46   372.43   0.48  53430.59

# LSTM val

40.72749999999999
9210.574215
1002.07
404.9982767250632
0.9562626297832763
47861.538613509474

# TOKEN val

29.230000000000004
9328.213329999999
697.13875
27.80061414688732
0.8242454638666243
48393.122216060154

# GMM fp val

In [45]:
pd.DataFrame.from_dict(res, orient='index', columns=['avg_rank', 'mae', 'mdae', 'mape', 'mdape', 'RMSE'])

avg_rank       mae    mdae      mape  mdape      RMSE
50      34.72  15613.16  721.75  29024.00   0.82  60424.81
100     24.21  16010.02  487.00  56976.87   0.73  67254.38
200     25.17  15689.47  539.62  56905.35   0.74  64796.85
300     25.34  11394.53  455.99    246.47   0.75  52871.75
500     25.80  13156.21  443.44    153.57   0.64  60751.96

# GMM rd val

In [196]:
pd.DataFrame.from_dict(res, orient='index', columns=['avg_rank', 'mae', 'mdae', 'mape', 'mdape', 'RMSE'])

avg_rank           mae     mdae     mape  mdape          RMSE
50     142.55  5.200131e+08   890.12  1028.06   0.99  1.030817e+10
100    141.65  5.200132e+08  1113.00  1049.12   0.99  1.030817e+10
200    133.04  5.200131e+08  1015.60  1046.03   0.99  1.030817e+10
300    131.76  5.200133e+08  1206.50  1062.09   0.99  1.030817e+10
500    129.76  5.200130e+08  1183.50  1054.86   0.99  1.030817e+10

# test

prototypes-300-alpha-1.5: { 97.73	4708165.84	505.50	33942.83	0.84	43280096.91}
gmm-fp-500: {95.70	4106513.28	447.11	23395.61	0.83	40715952.18}
gmm-rd-500: {91.90	5992735.71	377.75	1.052384e+05	0.71	46473984.89}
gmm-rd-100: {98.75	7300369.88	462.74	1.096382e+08	0.79	47641025.63}
LSTM: {178.420625 2227226.2244027485 1235.1 377.05618139245587 0.9885504173077713 34890274.7795075}
TOKEN: {111.11000000000001 4932059.560821875 15090.875 335922.118774415 8.135619093147778 42156167.30373767}
RANDOM: {200.5 2876568.267060938 1234.6724125 14129.0955220375 0.993455292191407 35124576.941615134}

Score1
prototypes-300-alpha-1.0: { 87.78	3869570.98	292.50	29742.22	0.57	42454933.58}
gmm-fp-500: {95.70	4106513.28	447.11	23395.61	0.83	40715952.18}
gmm-rd-300: {79.84	8760362.82	287.97	55090.97	0.57	64828718.70}
LSTM: {177.3375 3794685.5146099995 1477.5 126890.85061404701 0.9856678182097344 32143354.150388736}
TOKEN: {91.6275 3799029.317438 1235.55 24149.01943331 0.9922328236721986 24510576.80203356}
RANDOM: {200.5 2876568.267060938 1234.6724125 14129.0955220375 0.993455292191407 35124576.941615134}


score1 total random
sz-300 106.63	580.38	186.00	27.51	0.92	939.99
108.0225
587.527758385
290.5
68.06542070116728
0.7705220620358235
937.565120678442

In [202]:
pd.DataFrame.from_dict(res, orient='index', columns=['avg_rank', 'mae', 'mdae', 'mape', 'mdape', 'RMSE'])

avg_rank           mae     mdae     mape  mdape          RMSE
0.5     151.14  5.200132e+08  1079.99   117.18   0.99  1.030817e+10
0.75    146.40  5.200132e+08  1051.47    24.84   0.98  1.030817e+10
1       144.72  5.200132e+08  1055.47    25.88   0.98  1.030817e+10
1.25    142.28  5.200129e+08   842.73  1020.51   0.99  1.030817e+10
1.5     141.99  5.200129e+08   842.00  1015.65   0.99  1.030817e+10
2.5     142.35  5.200131e+08   849.00  1027.43   0.99  1.030817e+10

In [204]:
res = {}
v = s
# for k, v in s.items():
#     print(k)

score = 4

avg_rank = np.array(v['avg_rank'])
if score == 5:
    avg_avg_rank = (np.mean(avg_rank[1:]))
else:
    avg_avg_rank = (avg_rank[1:][score - 1])

print(avg_avg_rank)

mae = np.array(v['mae'])
if score == 5:
    avg_mae = (np.mean(mae))
else:
    avg_mae = (mae[score - 1])

print(avg_mae)

mdae = np.array(v['mdae'])
if score == 5:
    avg_mdae = (np.mean(mdae))
else:
    avg_mdae = mdae[score -1]
print(avg_mdae)

mape = np.array(v['mape'])
if score == 5:
    avg_mape = (np.mean(mape))
else:
    avg_mape = mape[score -1]

print(avg_mape)

mdape = np.array(v['mdape'])
if score == 5:
    avg_mdape = (np.mean(mdape))
else:
    avg_mdape = mdape[score -1]
print(avg_mdape)

RMSE = np.array(v['RMSE'])
if score == 5:
    avg_RMSE = (np.mean(RMSE))
else:
    avg_RMSE = RMSE[score -1]
print(avg_RMSE)


195.2375
520110071.00604415
1220.36
46679.64361248104
0.9995418442903612
10308169911.715693


In [ ]:
Prototype-300-1.5

        avg_rank	mae	mdae	mape	mdape	RMSE
score1 79.46	6323791.64	369.5	75676.16	0.77	6.930247e+07
score2 81.39	6323791.64	321.00	59335.96	0.64	54798614.98
score3 101.92	1294210.74	477.50	559.28	0.96	24509653.03
score4 128.16	1292932.63	854.00	199.90	0.98	24509650.61

